code/src/components/product-card/ProductCard.test.tsx

```diff
@@ -54,4 +71,26 @@ describe("ProductCard component", () => {
     const productLink = screen.getByRole("link");
     expect(productLink).toHaveAttribute("href", "/");
   });
+
+  test("should call useState on mount", () => {
+    expect(mockSetState).toHaveBeenCalledWith(true);
+  });
+
+  test("Should call functions on cart icon click", () => {
+    const cartIcon = screen.getByRole("button");
+
+    fireEvent.click(cartIcon);
+
+    expect(mockSetState).toHaveBeenNthCalledWith(3, true);
```

Comment:
Why setState should be called 3 times here?

Reply:
Mui's IconButton calls useState inside on mount. That is why there is 3 but not 2, as it is expected. I added comment for this.